alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -236,6 +236,28 @@ class DocumentLevelMonitorRunner : MonitorRunner() {
                         }
                     }
 
+                    val fieldsToBeQueried = mutableSetOf<String>()
+
+                    for (it in queries) {
+                        if (it.queryFieldNames.isEmpty()) {
+                            fieldsToBeQueried.clear()
+                            logger.debug(
+                                "Monitor ${monitor.id} : " +
+                                    "Doc Level query ${it.id} : ${it.query} doesn't have queryFieldNames populated. " +
+                                    "Cannot optimize monitor to fetch only query-relevant fields. " +
+                                    "Querying entire doc source."
+                            )
+                            break
+                        }
+                        fieldsToBeQueried.addAll(it.queryFieldNames)
+                    }
```

Comment:
Minor - could put this an if with `monitorCtx.fetchOnlyQueryFieldNames` as well to avoid the unnecessary work. Work should be negligible though so not a big deal

Unrelated to above - is there a scenario where we would want to turn this filtering off? Adding the setting to toggle on/off does no harm IMO, but not sure that it's necessary

Reply:
Retaining the feature flag as it's a super admin permission setting only so it can't be flipped off easily

Reply:
moved logic for population of fieldsToBeQueried  into if block